src/libraries/Common/tests/System/Security/Cryptography/AlgorithmImplementations/AES/KeyWrapTests.cs

```diff
@@ -0,0 +1,908 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using Test.Cryptography;
+using Xunit;
+
+namespace System.Security.Cryptography.Encryption.Aes.Tests
+{
+    using Aes = System.Security.Cryptography.Aes;
+
+    [SkipOnPlatform(TestPlatforms.Browser, "Not supported on Browser")]
+    public sealed class KeyWrapTests_AesCreate_KeyProp : KeyWrapTests
+    {
+        protected override Aes CreateKey(byte[] key)
+        {
+            Aes aes = Aes.Create();
+            aes.Key = key;
+            return aes;
+        }
+    }
+
+    [SkipOnPlatform(TestPlatforms.Browser, "Not supported on Browser")]
+    public sealed class KeyWrapTests_AesCreate_SetKey : KeyWrapTests
+    {
+        protected override Aes CreateKey(byte[] key)
+        {
+            Aes aes = Aes.Create();
+            aes.SetKey(key);
+            return aes;
+        }
+    }
+
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public sealed class KeyWrapTests_AesCryptoServiceProvider : KeyWrapTests
+    {
+        protected override Aes CreateKey(byte[] key)
+        {
+            Aes aes = new AesCryptoServiceProvider();
+            aes.Key = key;
+            return aes;
+        }
+    }
+
+    [PlatformSpecific(TestPlatforms.Windows)]
+    public sealed class KeyWrapTests_AesCng : KeyWrapTests
+    {
+        protected override Aes CreateKey(byte[] key)
+        {
+            Aes aes = new AesCng();
+            aes.Key = key;
+            return aes;
+        }
+    }
+
+    public static class KeyWrapContractTests
+    {
+        [Theory]
+        [InlineData(1, 16)]
+        [InlineData(5, 16)]
+        [InlineData(8, 16)]
+        [InlineData(9, 24)]
+        [InlineData(15, 24)]
+        [InlineData(16, 24)]
+        [InlineData(17, 32)]
+        public static void VerifyGetPaddedLength(int inputLength, int expectedLength)
+        {
+            Assert.Equal(expectedLength, Aes.GetKeyWrapPaddedLength(inputLength));
+        }
+
+        [Fact]
+        public static void VerifyGetPaddedLength_Random()
+        {
+            int value = Random.Shared.Next(0x7FFF_FFF1);
+            int actual = Aes.GetKeyWrapPaddedLength(value);
+
+            // Rather than repeat the formula of `(value + 7) / 8 * 8 + 8`, let's prove it by deconstruction.
+            int minus8 = actual - 8;
+            int minus16 = actual - 16;
+
+            if (value <= minus16 || value > minus8)
```

Comment:
There is `Assert.InRange` if you want to use it, but note that the upper and lower bounds are inclusive.

Reply:
I thought about switching to that... but then decided that `Assert.InRange(value, minus16 + 1, minus8)` looks... very non-obvious.